### PR TITLE
Adding BigQuery reader implementation and upgrading google-cloud-cpp and grpc libraries

### LINKED ISSUE #29: Support BigQuery for tensorflow-io
BigQuery is Google's serverless cloud data analytics platform. During last SIG I/O call the support for BigQuery (`BigQueryDataset`) was mentioned.

It would be good to add BigQuery (`BigQueryDataset`) to tensorflow-io so that users could use tensorflow to do more big data analytics on cloud.

Note that Google's BigQuery seems to have no C++ client library. It does have a python support and a RESTful API support. The implementation likely will need to use BigQuery's python client library, or http?

### PR DESCRIPTION
**Pull request by vlasenkoalexey**
Fixes #29
@sshrdp FYI

### CODE REVIEW

File: tensorflow_io/bigquery/ops/bigquery_ops.cc [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    .Attr("selected_fields: list(string) >= 1")
+    .Attr("output_types: list(type) >= 1")
+    .Attr("requested_streams: int")
+    .Attr("row_restriction: string = ''")
+    .Attr("container: string = ''")
+    .Attr("shared_name: string = ''")
+    .Output("streams: string")
+    .Output("avro_schema: string")
+    .SetShapeFn([](tensorflow::shape_inference::InferenceContext* c) {
+      c->set_output(0, c->Vector(1));
```
Comment (sshrdp) [inline tensorflow_io/bigquery/ops/bigquery_ops.cc]: I think It should be Unknown (c->Vector(c->UnknownDim())).

File: tensorflow_io/bigtable/kernels/bigtable_kernels.cc [hunk 2]
```diff
@@ -212,7 +213,7 @@ class ToBigtableOp : public AsyncOpKernel {
       column_families.reserve(column_families_tensor->NumElements());
       std::vector<string> columns;
       columns.reserve(column_families_tensor->NumElements());
-      for (uint64 i = 0; i < column_families_tensor->NumElements(); ++i) {
+      for (auto i = 0; i < column_families_tensor->NumElements(); ++i) {
```
[the remark this answers is not in the document]
Comment (vlasenkoalexey) [reply tensorflow_io/bigtable/kernels/bigtable_kernels.cc]: nope, reverted

File: tensorflow_io/bigquery/kernels/test_kernels/fake_bigquery_storage_service.h [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_IO_BIGQUERY_KERNELS_TEST_KERNELS_BIGTABLE_FAKE_H_
+#define TENSORFLOW_IO_BIGQUERY_KERNELS_TEST_KERNELS_BIGTABLE_FAKE_H_
+
+#include <cstring>
+#include <iostream>
```
[the remark this answers is not in the document]
Comment (vlasenkoalexey) [reply tensorflow_io/bigquery/kernels/test_kernels/fake_bigquery_storage_service.h]: removed

File: tensorflow_io/bigquery/ops/bigquery_ops.cc (hunk 1 above)
Comment (vlasenkoalexey) [reply tensorflow_io/bigquery/ops/bigquery_ops.cc]: updated and confirmed that it works as expected

**Comment by sshrdp** [comment]
@yongtang Yong, can you review this change?

**Pull request merged**